app/validation/validator.py

```diff
@@ -745,28 +755,29 @@ def _validate_checkbox_exclusive_conditions_in_when_rule(self, when, answer_ids_
 
         return errors
 
-    def _validate_comparison_id_in_when_rule(self, when, answer_ids_with_group_id, referenced_id):
+    def _validate_comparison_in_when_rule(self, when, answer_ids_with_group_id, referenced_id):
         """
         Validate that conditions requiring list match values define a comparison answer id that is of type Checkbox
         and ensure all other conditions with comparison id match answer types
         :return: list of dictionaries containing error messages, otherwise it returns an empty list
         """
         errors = []
 
-        answer_id, comparison_id, condition = when['id'], when['comparison_id'], when['condition']
-        comparison_answer_type = answer_ids_with_group_id[when['comparison_id']]['answer']['type']
-        id_answer_type = answer_ids_with_group_id[when['id']]['answer']['type']
-        conditions_requiring_list_match_values = ('equals any', 'not equals any', 'contains any', 'contains all')
+        if when['comparison']['source'] == 'answers':
+            answer_id, comparison_id, condition = when['id'], when['comparison']['id'], when['condition']
```

Comment:
This would be more readable if the assigned variables were used in the following mapping accessors